os/board/common/crashdump.c

```diff
@@ -162,11 +176,88 @@ static int ramdump_via_uart(void)
 }
 #endif
 
+#if defined(CONFIG_BOARD_SMARTFS_DUMP)
+static void print_sector(char *buf, int size) {
+	for (register int i = 0; i < size / 16; i++) {
+		for (register int j = 0; j < 16; j++) {
+			printf("%02x ", buf[i * 16 + j]);
+		}
+		printf("\n");
+	}
+}
+
+static int smartfs_dump(void)
+{
+	int i;
+	int ch;
+	char *handshake_str = SDV_HANDSHAKE_STRING;
+	char handshake_msg_buf[SDV_HANDSHAKE_STR_LEN + 1] = "";
+	int totalsectors = smartfsdump_partsize / CONFIG_MTD_SMART_SECTOR_SIZE;
+	int offset = CONFIG_FLASH_START_ADDR + smartfsdump_partoffset;
+
+#if !defined(CONFIG_ARCH_LOWPUTC)
+	/* If low-level serial is not available, smartfs_dump is not possible */
+	return -1;
+#endif
+
+	/* Inform the terminal user */
+	up_puts("\n************************************************************************************************\n");
+	up_puts("\t\tDisconnect this serial terminal and Run SDV Tool\n");
+	up_puts("************************************************************************************************\n");
+
+	/* Receive handshake string from HOST */
+	do {
+		handshake_msg_buf[0] = up_getc();
+	} while (handshake_msg_buf[0] != handshake_str[0]);
+
+	for (i = 1; i < strlen(handshake_str);) {
+		ch = up_getc();
+		if (ch != -1) {
+			handshake_msg_buf[i] = ch;
+			i++;
+		}
+	}
+	handshake_msg_buf[SDV_HANDSHAKE_STR_LEN] = '\n';
+
+	if (strncmp(handshake_msg_buf, handshake_str, strlen(handshake_str)) != 0) {
+		fdbg("\tHandshake MSG is wrong... len = %d, %s\n", strlen(handshake_str), handshake_msg_buf);
```

Comment:
Because this is in crash state, you should not concern log configuration. Let's use lldbg for necessary output.